Sources/MongoSwift/MongoClient.swift

```diff
@@ -253,21 +240,6 @@ public class MongoClient {
         return try sessionBody(session)
     }
 
-    /**
-     * Closes the client.
-     */
-    public func close() {
```

Comment:
this method (and `deinit`) can completely go away because:
- pool now handles cleaning up libmongoc client
- we don't need to explicitly disable monitoring anymore. I'm not even sure we needed to do it before. I think it was out of concern that some APM event might be in transit to us while a `MongoClient` got closed, and then we would still publish the event after it was closed. `disableMonitoring` makes it so that if receive a future APM event using this `MongoClient` we won't publish it. 
now that you can't close the `MongoClient` yourself the first part doesn't seem like a concern. Though if we receive an APM event after its relevant `MongoClient` (and therefore also the libmongoc client generating the callback) has been deinitialized, we would crash [here](https://github.com/mongodb/mongo-swift-driver/blob/ce7ad3abf520ee0494517499d07cf83b401263ca/Sources/MongoSwift/APM.swift#L469). (this has always been the case and I'm not sure that sequence of events is even possible?)

Reply:
It doesn't seem to me that the crash would be possible assuming we're accessing the underlying `mongoc_client_t` with only one thread.